add file watcher, importer module, csv to json converter

kartelpol opened the pull request:
Link to the task: https://epam-my.sharepoint.com/personal/vitali_kozlov_epam_com/_layouts/15/onedrive.aspx?id=%2Fpersonal%2Fvitali_kozlov_epam_com%2FDocuments%2FNode%2Ejs%2FCDP%2FHomeworks%2F2%2E%20Async%20development%20introduction%2FHomework%202%2Epdf&parent=%2Fpersonal%2Fvitali_kozlov_epam_com%2FDocuments%2FNode%2Ejs%2FCDP%2FHomeworks%2F2%2E%20Async%20development%20introduction&slrid=64b5ab9e-5099-7000-9a33-d45254a84c8a

Commit:
add file watcher, importer module, csv to json converter

deser commented on app.js:
```diff
@@ -1,9 +1,42 @@
+const watchModule = new Dirwatcher();
+const importModule = new Importer();
+
+watchModule.on('changed', (path) => { 
```
Is this code able to track **all** changed files since last check?

deser replied:
I see that it is. But variable name `path` is really confusing

deser commented on app.js:
```diff
@@ -1,9 +1,42 @@
+        .catch(err => logger.error(err));
+
+    const importedData = importModule.importSync(path);
+    //console.log(importedData);
```
remove :)

deser commented on config/logger.js:
```diff
@@ -1,11 +1,11 @@
 
 const logger = createLogger({
-    format : format.printf(info => info.message),
+    format : format.printf(info => typeof info.message === 'object' ? JSON.stringify(info.message) : info.message),
```
you can use pretty formatting of JSON.stringify

deser commented on log:
```diff
@@ -0,0 +1,3082 @@
+Error Error reading ./data
```
add log to gitignore

deser commented on src/dirwatcher.js:
```diff
@@ -0,0 +1,77 @@
+        this.modificationDates = {};
+    }
+
+    watch(path, delay) {   
```
comments\jsdocs to every function would be appreciated

deser commented on src/dirwatcher.js:
```diff
@@ -0,0 +1,77 @@
+    }
+
+    watch(path, delay) {   
+        const timeout = setInterval(()=> {
```
it might make sense to return interval ID from watch function to clean up resources when needed.

deser commented on src/dirwatcher.js:
```diff
@@ -0,0 +1,77 @@
+
+            Promise.all(promisifiedStats).then(() => {
+                if (this.modifiedFiles.length) {
+                    this.emitEvent('changed', this.modifiedFiles);
```
modifiedFiles could be confusing. Better to rename to smth like `pathsOfModifiedFiles`

deser commented on app.js:
```diff
@@ -1,9 +1,42 @@
+const importModule = new Importer();
+
+watchModule.on('changed', (path) => { 
+    importModule.import(path).then(dataArr => {
```
Also, a small notice is that according to task `import` method should be able to work with only 1 path, not with array

deser commented on src/importer.js:
```diff
@@ -0,0 +1,37 @@
+import * as util from 'util';
+import logger from '../config/logger.js';
+
+class importer {
```
follow notation, `Improter`

deser commented on src/dirwatcher.js:
```diff
@@ -0,0 +1,77 @@
+import * as EventEmitter from 'events';
+import logger from '../config/logger';
+
+class dirwatcher extends EventEmitter {
```
follow notation, `Dirwatcher`

deser replied:
export default class ...

deser commented on src/importer.js:
```diff
@@ -0,0 +1,37 @@
+import logger from '../config/logger.js';
+
+class importer {
+    import(files) {
```
according to task it should be able to import only 1 file

deser commented on src/importer.js:
```diff
@@ -0,0 +1,37 @@
+        return Promise.all(modifiedFiles);
+    }
+
+    importSync(files) {
```
according to task it should be able to import only 1 file

deser commented on src/importer.js:
```diff
@@ -0,0 +1,37 @@
+                modifiedData.push(data);
+            } 
+            catch(err) {
+                logger.error(`Error reading ${filename}`);
```
I believe winston allows you to put `err` object here and string with message as second param

Commit:
apply review comments
	add jsdoc
	fix import module due to task (takes and import only one file)
	prettify logging and naming
	add logs to the gitignore
